x/auth/src/client/cli/tx_query.rs

```diff
@@ -0,0 +1,390 @@
+                }
+                if txs.txs.len() > 1 {
+                    return Err(anyhow!(
+                        "found {} txs matching given parameters",
+                        txs.txs.len()
+                    ));
+                }
+                Ok(Self::QueryResponse::Txs(QueryTxsResponse { txs }))
+            }
+        }
+    }
+}
+
+// to have full match of cli in app
+// need to be added as a member of client cli
+/// Query for paginated transactions that match a set of events
+#[derive(Args, Debug)]
+pub struct TxsQueryCli {
+    #[arg(long)]
+    pub events: String,
+    #[arg(long, default_value_t = 1)]
+    pub page: u32,
+    #[arg(long, default_value_t = 30)]
+    pub limit: u32,
+}
+
+#[derive(Debug, Clone)]
+pub struct TxsQueryHandler<M: TxMessage> {
+    msgs_type: PhantomData<M>,
+}
+
+impl<M: TxMessage> TxsQueryHandler<M> {
+    /// Constructor allows to handle tx messages received by query calls.
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> TxsQueryHandler<M> {
+        TxsQueryHandler {
+            msgs_type: PhantomData,
+        }
+    }
+}
+
+impl<M: TxMessage> QueryHandler for TxsQueryHandler<M> {
+    type QueryRequest = TxQuery;
+
+    type QueryCommands = TxsQueryCli;
+
+    type QueryResponse = TxQueryResponse<M>;
+
+    fn prepare_query_request(
+        &self,
+        command: &Self::QueryCommands,
+    ) -> anyhow::Result<Self::QueryRequest> {
+        let TxsQueryCli {
+            events,
+            page,
+            limit,
+        } = command;
+
+        let events_str = events.trim_start_matches('\'').trim_end_matches('\'');
+        let events = events_str.split('&');
+
+        let mut tm_events = Vec::with_capacity(events_str.matches('&').count() + 1);
+        for event in events {
+            if !event.contains('=') || event.matches('=').count() > 1 {
+                return Err(anyhow!("invalid event; event {event} should be of the format: {{eventType}}.{{eventAttribute}}={{value}}"));
+            }
+            let tokens = event.split('=').collect::<Vec<_>>();
+            if tokens[0] == "tx.height" {
+                tm_events.push(format!("{}={}", tokens[0], tokens[1]));
+            } else {
+                tm_events.push(format!("{}='{}'", tokens[0], tokens[1]));
+            }
+        }
+        Ok(Self::QueryRequest::Txs(QueryGetTxsEventRequest {
+            events: tm_events,
+            order_by: "asc".to_string(),
+            page: *page,
+            limit: *limit,
+        }))
+    }
+
+    fn execute_query_request(
+        &self,
+        query: Self::QueryRequest,
+        node: url::Url,
+        _height: Option<gears::tendermint::types::proto::block::Height>,
+    ) -> anyhow::Result<Vec<u8>> {
+        let client = HttpClient::new(node.as_str())?;
+        let runtime = tokio::runtime::Runtime::new().expect("failed to create tokio runtime");
+
+        match query {
+            Self::QueryRequest::Tx(_) => unreachable!(),
```

Comment:
unreachable?

Reply:
I'm reusing enum for requests and this variant is not used in the command.

Reply:
fixed